src/Sylius/Component/Product/Factory/ProductFactory.php

```diff
@@ -72,6 +78,26 @@ public function createNew()
     /**
      * {@inheritdoc}
      */
+    public function createTyped($type)
+    {
+        Assert::oneOf($type, [ProductInterface::SIMPLE, ProductInterface::CONFIGURABLE]);
```

Comment:
Do we even need this asserton here, if product itself has the same on `setType` method?